packages/ui/src/components/Pagination/SPagination.vue

```diff
@@ -0,0 +1,377 @@
+<script setup lang="ts">
+import {
+  IconArrowsChevronRightXs24,
+  IconArrowsChevronLeftXs24,
+  IconChevronsLeft16,
+  IconChevronsRight16,
+} from '@/components/icons'
+import { not } from '@vueuse/core'
+import { SDropdown } from '@/components'
+
+const PAGINATION_MAX_PAGES_SELECTABLE = 7
+const PAGINATION_JUMP_SIZE = PAGINATION_MAX_PAGES_SELECTABLE - 2
+const PAGINATION_BREAKPOINT_WIDTH = 480
+
+type PaginationJumpValues = number | 'jumpNext' | 'jumpPrev'
+
+const props = withDefaults(
+  defineProps<{
+    /**
+     * total item count
+     * */
+    total?: number
+    /**
+     * item count of each page, two-way bound
+     * */
+    pageSize?: number | null
+    /**
+     * current page number, two-way bound
+     * */
+    currentPage?: number
+    /**
+     * options of item count per page
+     * */
+    pageSizes?: number[]
+    /**
+     * text shown near sizes dropdown
+     * */
+    sizesLabel?: string
+  }>(),
+  {
+    total: 0,
+    pageSize: null,
+    currentPage: 1,
+    pageSizes: () => [10, 50, 100],
+    sizesLabel: 'Rows per page',
+  },
+)
+/* eslint-disable @typescript-eslint/unified-signatures */
+const emit = defineEmits<{
+  (event: 'size-change', value: number): void
+  (event: 'current-change', value: number): void
+  (event: 'prev-click', value: number): void
+  (event: 'next-click', value: number): void
+  (event: 'update:currentPage', value: number): void
+  (event: 'update:pageSize', value: number): void
+}>()
+
+const current = useVModel(props, 'currentPage', emit, { passive: true })
+const size = useVModel(props, 'pageSize', emit, { passive: true })
+const numericSize = computed({
+  get: () => {
+    return size.value ?? (props.pageSizes[0] || 10)
+  },
+  set: (value: null | number) => {
+    size.value = value
+  },
+})
+
+const sizeOptions = computed(() => props.pageSizes.map((size) => ({ label: String(size), value: size })))
+
+const pagesNum = computed(() => Math.max(1, Math.ceil(props.total / numericSize.value)))
+const isCurrentFirst = computed(() => current.value === 1)
+const isCurrentLast = computed(() => current.value === pagesNum.value)
+const isAbleJumpPrev = computed(() => current.value >= PAGINATION_JUMP_SIZE)
+const isAbleJumpNext = computed(() => current.value <= pagesNum.value - PAGINATION_JUMP_SIZE + 1)
+const shouldShowControls = computed(() => pagesNum.value > PAGINATION_MAX_PAGES_SELECTABLE)
+const firstItemNum = computed(() => (current.value - 1) * numericSize.value + 1)
+const lastItemNum = computed(() => Math.min(props.total, current.value * numericSize.value))
```

Comment:
Maybe it makes sense to ensure that there are no unnecessary re-renders caused by these set of computeds, and maybe use `eagerComputed` here instead.